### PR TITLE
Dev rc

### PR DESCRIPTION
**Pull request by kripsy**
https://app.codecov.io/gh/kripsy/GophKeeper/tree/dev-rc

### CODE REVIEW

**Comment by kripsy** [comment]
Это первый коммит.
Далее будут доделки клиентской части, юнит тесты, документация и "озеленение" git actions.

File: cmd/testServer/main.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	"log"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"google.golang.org/grpc"
+)
+
+const BUCKETNAME = "secrets"
```
Comment (arttet) [inline cmd/testServer/main.go:17]: Please avoid secrets in the code database

File: cmd/testServer/main.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	// if err != nil {
+	// 	log.Fatalf("Failed to receive response: %v", err)
+	// }
+	// log.Printf("Response: %s", resp.Status)
+
+	// Для демонстрации: ожидание, чтобы увидеть, как "keepalive" работает
+	time.Sleep(3000 * time.Second)
+}
+
+func main_minio() {
```
Comment (arttet) [inline cmd/testServer/main.go:123]: Please rename it

File: Dockerfiles/server/Dockerfile [hunk 3]
```diff
@@ -11,13 +11,16 @@ RUN go mod download
...
-COPY ./docs/server ./docs/server
 COPY ./internal/server ./internal/server
-COPY ./internal/common ./internal/common
+COPY ./internal/utils ./internal/utils
+COPY ./internal/logger ./internal/logger
+COPY ./internal/models ./internal/models
+COPY ./pkg ./pkg
+COPY ./gen ./gen
+COPY ./cert ./cert
 COPY ./*.md ./
```
Comment (arttet) [inline Dockerfiles/server/Dockerfile]: Please optimize Dockerfile

Comment (arttet) [reply Dockerfiles/server/Dockerfile]: https://docs.docker.com/build/building/multi-stage/

File: MakeFile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+build/client/linux:
+	go build -o ./cmd/client/client \
+	-ldflags "-X main.buildVersion=$(CLIENT_VERSION) \
+		  -X main.buildDate=$(shell date '+%Y/%m/%d[%H:%M:%S]')" \
+	./cmd/client/*.go
+
+
+build/client/windows:
+	GOOS=windows GOARCH=amd64 go build -o ./cmd/client/client.exe  \
```
Comment (arttet) [inline MakeFile]:
```suggestion
	GOOS=windows GOARCH=amd64 go build -o ./cmd/client/client$(shell go env GOEXE)  \
```

File: MakeFile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+run/server:
+	go run ./cmd/server/main.go
+
+run/client:
+	go run ./cmd/client/
+
+#-secure=false
+
+build/client/linux:
```
Comment (arttet) [inline MakeFile:13]: What about macOS?

File: MakeFile (hunk 4 above)
Comment (arttet) [inline MakeFile]: Do you use cross compiling?

File: internal/client/infrastrucrure/ui/cli/delete_secret.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package cli
+
+import (
+	"github.com/kripsy/GophKeeper/internal/client/infrastrucrure/ui"
+	"github.com/kripsy/GophKeeper/internal/models"
+	"github.com/manifoldco/promptui"
+)
+
+func (c CLI) DeleteSecret(metaData models.MetaData) (string, bool) {
```
Comment (arttet) [inline internal/client/infrastrucrure/ui/cli/delete_secret.go]:
```suggestion
func (c *CLI) DeleteSecret(metaData models.MetaData) (string, bool) {
```

File: internal/client/infrastrucrure/ui/cli/helpers.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+	dateFormat = "[2006/01/02 15:01]"
+)
+
+func (c CLI) checkInterrupt(err error) {
+	if errors.Is(err, promptui.ErrInterrupt) {
+		c.Exit()
+	}
+}
+
+var clearMap map[string]func() error
```
Comment (arttet) [inline internal/client/infrastrucrure/ui/cli/helpers.go]: Please rename it

File: internal/client/infrastrucrure/filemanager/userauth.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	"github.com/kripsy/GophKeeper/internal/models"
+	"github.com/kripsy/GophKeeper/internal/utils"
+)
+
+const (
+	fileMode os.FileMode = 0700 //660
+	dirMode  os.FileMode = 0700 //755
+)
+
+type UserAuth struct {
```
Comment (arttet) [inline internal/client/infrastrucrure/filemanager/userauth.go]: It should be a private struct

**Review by arttet — CHANGES_REQUESTED** [review]

File: cmd/testServer/main.go (hunk 1 above)
Comment (kripsy) [reply cmd/testServer/main.go:17]: it's not a secret. 
in addition, testServer/ is a development and debugging package. does not carry a functional load for services

File: cmd/testServer/main.go (hunk 2 above)
Comment (kripsy) [reply cmd/testServer/main.go:123]: testServer/ is a development and debugging package. does not carry a functional load for services

File: Dockerfiles/server/Dockerfile (hunk 3 above)
Comment (kripsy) [reply Dockerfiles/server/Dockerfile]: done
https://github.com/kripsy/GophKeeper/pull/15/commits/8634678031acaaba4ae49979318a05febdd032e0

File: MakeFile (hunk 4 above)
Comment (kripsy) [reply MakeFile]: @Ostrovsky42 fix pls

File: MakeFile (hunk 5 above)
Comment (kripsy) [reply MakeFile:13]: build for macos added by @Ostrovsky42

File: MakeFile (hunk 4 above)
Comment (kripsy) [reply MakeFile]: @arttet 
it's actual?
for cross compiling we use set env GOOS and GOARCH

File: internal/client/infrastrucrure/ui/cli/delete_secret.go (hunk 6 above)
Comment (kripsy) [reply internal/client/infrastrucrure/ui/cli/delete_secret.go]: @Ostrovsky42 fix pls

File: internal/client/infrastrucrure/ui/cli/helpers.go (hunk 7 above)
Comment (kripsy) [reply internal/client/infrastrucrure/ui/cli/helpers.go]: @Ostrovsky42 fix pls

File: internal/client/infrastrucrure/filemanager/userauth.go (hunk 8 above)
Comment (kripsy) [reply internal/client/infrastrucrure/filemanager/userauth.go]: @Ostrovsky42 fix pls

**Comment by arttet** [comment]
Please add the link to the `CodeCov` https://app.codecov.io/gh/kripsy/GophKeeper/tree/dev-rc in this description.

**Comment by kripsy** [comment]
@arttet 
80.14% coverage 
https://app.codecov.io/gh/kripsy/GophKeeper/tree/dev-rc

**Pull request merged**